plugins/opencritic.py

```diff
@@ -30,17 +33,99 @@ def oc(text):
         if item['name'] == bestChoice:
             gameId = item['id']
             gameTitle = item['name']
-            gameUrl = 'http://opencritic.com/game/' + str(gameId) + '/' + re.sub('[^0-9a-zA-Z]+', '-', item['name']).lower()
+            gameUrl = 'http://opencritic.com/game/' + \
+                str(gameId) + '/' + \
+                re.sub('[^0-9a-zA-Z]+', '-', item['name']).lower()
+
+    games.append(gameId)
 
     scoreQuery = {
-        'id':gameId
+        'id': gameId
     }
+    scoreResponse = requests.request("GET", scoreUrl, headers=headers, params=scoreQuery).json()
 
-    response = requests.request("GET", scoreUrl, headers = headers, params = scoreQuery).json()
+    dateQuery = '{"ids": ' + str(games) + ', "orderBy": "timeAscending"}'
+    gameDate = requests.post(dateUrl, headers=headers, data=dateQuery).json()[0]['releaseDate'][:10]
 
-    try: 
-        gameScore = round(float(response['score']))
+    try:
+        gameScore = round(float(scoreResponse['score']))
     except:
-        return '\x02{}\x02 does not have an average score yet.'.format(gameTitle)
+        return '\x02{}\x02 releases on {}.'.format(gameTitle, gameDate)
     else:
         return '\x02{}\x02 - \x02Score:\x02 {} - {}'.format(gameTitle, gameScore, gameUrl)
+
+@hook.command('octop')
+def octop(text):
+    searchUrl = url + 'filter'
+    output = []
+    startDate = datetime.datetime.now().year
+
```

Comment:
you could store text.lower() in a variable here for the references below. That will cut down on all the extra calls.

Reply:
actually, why not make this check a separate function, then you can reduce code duplication as well.

Reply:
I like both of these ideas--still learning, but while I understand the concept of calling a function inside a function to return the called function's value, I'm not sure what that would look like from an implementation standpoint for the purposes of being called via a command in chat.

Essentially, if a user does !ocup/!octop text, we would store text.lower() during x function and then immediately run the y function associated with the command, referencing the original function that stored text.lower()?

Reply:
nah no need for all of that. you can just do something like
```python
def check_the_things(in_text):
    in_text = in_text.lower()
    if in_text .... {do all your checks}

def my_function(command):
    value_desired = check_the_things(command.text)
    ... do something with value_desired
```

From a user perspective, it is no different to them at all, since the lowering can all be handled inside the external function.